uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/select/SelectComponent.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.uberfire.ext.widgets.common.client.select;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.jboss.errai.ioc.client.api.ManagedInstance;
+import org.uberfire.client.mvp.UberElemental;
+
+public class SelectComponent {
+
+    private View view;
+    private ManagedInstance<SelectOptionComponent> selectOptionWidget;
+    private SelectOption selected;
+    private Consumer<SelectOption> callback;
+    private List<SelectOptionComponent> components;
+
+    @Inject
+    public SelectComponent(final View view,
+                           final ManagedInstance<SelectOptionComponent> selectOptionComponent) {
+        this.view = view;
+        this.selectOptionWidget = selectOptionComponent;
+    }
+
+    @PostConstruct
+    public void init() {
+        this.view.init(this);
+    }
+
+    public void setup(List<SelectOption> options,
+                      Consumer<SelectOption> selectCallback) {
+        this.setSelected(options.get(0));
```

Comment:
Is this intended to be generic SelectComponent? If yes and somebody provides empty list of options, you're getting `ArrayIndexOutOfBoundsException`  here ..

Also a test with empty options list would be nice.